### PR TITLE
MINOR: remove old procesor API MockInternalProcessorContext

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/state/internals/TimeOrderedKeyValueBufferTest.java [hunk 1]
```diff
@@ -1049,11 +1048,12 @@ private static void putRecord(final TimeOrderedKeyValueBuffer<String, String, Ch
         buffer.put(streamTime, new Record<>(key, new Change<>(value, null), 0L), recordContext);
     }
 
+    @SuppressWarnings("resource")
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/TimeOrderedKeyValueBufferTest.java:1051]: Side cleanup

File: streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBBlockCacheMetricsTest.java [hunk 2]
```diff
@@ -79,17 +80,22 @@ static void withStore(final RocksDBStore store, final StateStoreContext context,
     @ParameterizedTest
     @MethodSource("stores")
     public void shouldRecordCorrectBlockCacheCapacity(final RocksDBStore store, final StateStoreContext ctx) {
-        withStore(store, ctx, () ->
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBBlockCacheMetricsTest.java:82]: side cleanup

File: streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBBlockCacheMetricsTest.java [hunk 3]
```diff
@@ -79,17 +80,22 @@ static void withStore(final RocksDBStore store, final StateStoreContext context,
...
     @MethodSource("stores")
     public void shouldRecordCorrectBlockCacheUsage(final RocksDBStore store, final StateStoreContext ctx) {
         withStore(store, ctx, () -> {
             final BlockBasedTableConfigWithAccessibleCache tableFormatConfig = (BlockBasedTableConfigWithAccessibleCache) store.getOptions().tableFormatConfig();
-            final long usage = tableFormatConfig.blockCache().getUsage();
-            assertMetric(ctx, STATE_STORE_LEVEL_GROUP, RocksDBMetrics.USAGE_OF_BLOCK_CACHE, BigInteger.valueOf(usage));
+            try (final Cache blockCache = tableFormatConfig.blockCache()) {
+                final long usage = blockCache.getUsage();
+                assertMetric(ctx, STATE_STORE_LEVEL_GROUP, RocksDBMetrics.USAGE_OF_BLOCK_CACHE, BigInteger.valueOf(usage));
+            }
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBBlockCacheMetricsTest.java:98]: side cleanup

File: streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBBlockCacheMetricsTest.java [hunk 4]
```diff
@@ -98,11 +104,14 @@ public void shouldRecordCorrectBlockCacheUsage(final RocksDBStore store, final S
     public void shouldRecordCorrectBlockCachePinnedUsage(final RocksDBStore store, final StateStoreContext ctx) {
         withStore(store, ctx, () -> {
             final BlockBasedTableConfigWithAccessibleCache tableFormatConfig = (BlockBasedTableConfigWithAccessibleCache) store.getOptions().tableFormatConfig();
-            final long usage = tableFormatConfig.blockCache().getPinnedUsage();
-            assertMetric(ctx, STATE_STORE_LEVEL_GROUP, RocksDBMetrics.PINNED_USAGE_OF_BLOCK_CACHE, BigInteger.valueOf(usage));
+            try (final Cache blockCache = tableFormatConfig.blockCache()) {
+                final long usage = blockCache.getPinnedUsage();
+                assertMetric(ctx, STATE_STORE_LEVEL_GROUP, RocksDBMetrics.PINNED_USAGE_OF_BLOCK_CACHE, BigInteger.valueOf(usage));
+            }
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBBlockCacheMetricsTest.java:110]: side cleanup

File: streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBBlockCacheMetricsTest.java [hunk 5]
```diff
@@ -98,11 +104,14 @@ public void shouldRecordCorrectBlockCacheUsage(final RocksDBStore store, final S
...
         });
     }
 
+    @SuppressWarnings("resource")
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBBlockCacheMetricsTest.java:114]: side cleanup

File: streams/src/test/java/org/apache/kafka/test/MockInternalNewProcessorContext.java [hunk 6]
```diff
@@ -41,12 +41,16 @@
...
+import java.util.LinkedHashMap;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Properties;
 
 public class MockInternalNewProcessorContext<KOut, VOut> extends MockProcessorContext<KOut, VOut> implements InternalProcessorContext<KOut, VOut> {
 
-    private ProcessorNode currentNode;
+    private ProcessorNode<?, ?, ?, ?> currentNode;
+    private RecordCollector recordCollector;
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/test/MockInternalNewProcessorContext.java:52]: Add missing `recordCollector` "feature -- the old `MockInternalProcessorContext` that we remove with this PR had this, and the test we rewrite use this feature

File: streams/src/test/java/org/apache/kafka/test/MockInternalNewProcessorContext.java [hunk 7]
```diff
@@ -41,12 +41,16 @@
...
+    private final Map<String, StateRestoreCallback> restoreCallbacks = new LinkedHashMap<>();
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/test/MockInternalNewProcessorContext.java:53]: Same

File: streams/src/test/java/org/apache/kafka/test/MockInternalNewProcessorContext.java [hunk 8]
```diff
@@ -108,12 +112,12 @@ public void setHeaders(final Headers headers) {
     }
 
     @Override
-    public void setCurrentNode(final ProcessorNode currentNode) {
+    public void setCurrentNode(final ProcessorNode<?, ?, ?, ?> currentNode) {
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/test/MockInternalNewProcessorContext.java:115]: side cleanup

**Review by bbejeck — APPROVED** [review]
LGTM

**Pull request merged**